presto-main/src/main/java/com/facebook/presto/server/SerializedPageWriteListener.java

```diff
@@ -78,6 +77,14 @@ public void onWritePossible()
                 bufferPosition += SIZE_OF_INT;
                 slice.setInt(bufferPosition, page.getSizeInBytes());
                 bufferPosition += SIZE_OF_INT;
+                if (CHECKSUMMED.isSet(page.getPageCodecMarkers())) {
+                    slice.setLong(bufferPosition, page.getChecksum());
+                    bufferPosition += SIZE_OF_LONG;
+                    slice.setInt(bufferPosition, page.getHostName().length());
```

Comment:
In theory it may contain non latin character (unlikely, but possible). Ideally it would be best if we could cache this value somewhere as byte array to avoid re-encoding for every page